### PR TITLE
remove the ipvs checks from the preflight checks

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What type of PR is this?**

/sig cluster-lifecycle
/kind cleanup
/priority important-soon

**What this PR does / why we need it**: removes the ipvs checks from kubeadm prelight checks

**Which issue(s) this PR fixes**: part of https://github.com/kubernetes/kubeadm/issues/1600

**Special notes for your reviewer**:

/assign @neolit123 @rosti 

**Does this PR introduce a user-facing change?**:

```release-note
kubeadm no longer performs IPVS checks as part of its preflight checks
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/preflight/checks_linux.go [hunk 1]
```diff
@@ -42,14 +41,4 @@ func (idsc IsDockerSystemdCheck) Check() (warnings, errorList []error) {
...
-// Check determines if IPVS proxier can be used or not
-func (ipvspc IPVSProxierCheck) Check() (warnings, errors []error) {
-	ipsetInterface := utilipset.New(ipvspc.exec)
-	kernelHandler := ipvs.NewLinuxKernelHandler()
-	if _, err := ipvs.CanUseIPVSProxier(kernelHandler, ipsetInterface); err != nil {
-		return nil, []error{err}
-	}
-	return nil, nil
-}
+}
```
Comment (neolit123) [inline cmd/kubeadm/app/preflight/checks_linux.go]: missing new line at end of file.

File: cmd/kubeadm/.import-restrictions [hunk 2]
```diff
@@ -79,7 +79,6 @@
 				"k8s.io/kubernetes/pkg/util/hash",
 				"k8s.io/kubernetes/pkg/util/ipset",
 				"k8s.io/kubernetes/pkg/util/iptables",
-				"k8s.io/kubernetes/pkg/util/ipvs",
```
Comment (neolit123) [inline cmd/kubeadm/.import-restrictions]: i don't know if import-boss allows us to find redundant allowed imports. i think not.
but i was under the impression that by removing this one we are no longer going to need a bulk of other imports.

**Comment by neolit123** [comment]
thanks @yastij 
added a couple of comments.

/approve

File: cmd/kubeadm/.import-restrictions (hunk 2 above)
Comment (yastij) [reply cmd/kubeadm/.import-restrictions]: Yes, I'm listing the import tree

Comment (neolit123) [reply cmd/kubeadm/.import-restrictions]: do you mean you verified that we can only remove this import?

**Comment by neolit123** [comment]
btw maybe we should wait before https://github.com/kubernetes/kubernetes/pull/81752 merges as probably the import-restrictions file for kubeadm will conflict and that is a much bigger PR.

File: cmd/kubeadm/.import-restrictions (hunk 2 above)
Comment (yastij) [reply cmd/kubeadm/.import-restrictions]: we can remove the 3 that this PR is removing

Comment (neolit123) [reply cmd/kubeadm/.import-restrictions]: ok, understood.
thanks

**Review by rosti — APPROVED** [review]
Thanks @yastij !
/lgtm

@neolit123 I doubt, that the import boss PR is going to merge any time soon. Let's see what happens over the weekend and merge this in the beginning of next week.

**Comment by neolit123** [comment]
/hold cancel
/lgtm

the related PR merged without .import-restriction changes in cmd/kubeadm.

**Pull request merged**